### DISCUSSION TITLE
Overwrite orphans too

### DISCUSSION THREAD

**Discussion by nedtwigg** [origin-energy/java-snapshot-testing#134] [Q&A]
I've got this in my `build.gradle`, works great...

```
test {
	if (System.env['CI'] == null) {
		systemProperty "updateSnapshot", ""
	}
}
```

.. except when I remove or rename tests, because I get orphan snapshot errors. Then I've got to manually mess around with `.snap` and `.snap.debug` files again.

Is there an `updateSnapshot` mode which also automatically removes orphans?

**Comment by jackmatt2** [comment]
I had an issue sitting around for a very long time concerning this exact issue
https://github.com/origin-energy/java-snapshot-testing/issues/45

I recall there being some difficulties though. When the application runs it has no context from where the source files are actually located.  I think maybe a separate tool would be needed to cleanup the orphan snapshot files.

**Reply by jackmatt2** [reply]
Oh, I think wha you are having trouble with is different from my above comment.  I would think that this should be possible.  In fact I thought that that was already happening.

**Reply by nedtwigg** [reply]
If you'd like an example, here's a commit where `./gradlew test` will fail with `ERROR: Found orphan snapshots` (even on a dev machine where `updateSnapshot` is set) 

- commit: https://github.com/equodev/equo-ide/commit/65d7524b7465b29116314564194e42efc57f15f6
- ci: https://github.com/equodev/equo-ide/actions/runs/3671391605/jobs/6206562411

And in the very next commit, ci will pass after removing the orphans

- commit: https://github.com/equodev/equo-ide/commit/0c143e6473c68d03b0273132a617991aa6274b15
- ci: https://github.com/equodev/equo-ide/actions/runs/3671393164/jobs/6206565193
